src/main/java/seedu/address/model/transaction/TransactionLog.java

```diff
@@ -106,4 +106,76 @@ public ObservableList<Transaction> getBuyTransactionList() {
         }
         return FXCollections.unmodifiableObservableList(internalList);
     }
+
+    /**
+     * Creates an observable list of transactions to be used by MainWindow.
+     * @return an unmodifiableObservableList of sorted transactions by oldest.
+     */
+    public ObservableList<Transaction> getOldestTransactionList() {
+        ObservableList<Transaction> internalList = FXCollections.observableArrayList();
+        internalList.addAll(transactionList);
+        quickSort(internalList, 0, internalList.size() - 1, 1);
+        return FXCollections.unmodifiableObservableList(internalList);
+    }
+
+    /**
+     * Creates an observable list of transactions to be used by MainWindow.
+     * @return an unmodifiableObservableList of sorted transactions by latest.
+     */
+    public ObservableList<Transaction> getLatestTransactionList() {
+        ObservableList<Transaction> internalList = FXCollections.observableArrayList();
+        internalList.addAll(transactionList);
+        quickSort(internalList, 0, internalList.size() - 1, 0);
+        return FXCollections.unmodifiableObservableList(internalList);
+    }
+
+    private static void swap(ObservableList<Transaction> arr, int i, int j) {
+        Transaction temp = arr.get(i);
+        arr.set(i, arr.get(j));
+        arr.set(j, temp);
+    }
```

Comment:
I think this method is out of place, and does not belong to `TransactionLog` class as it is not related to Transactions.
In fact, it probably wouldn't be needed if we use Java's Collection class.